libcontainer/container_linux.go

```diff
@@ -230,6 +230,13 @@ func (c *linuxContainer) Set(config configs.Config) error {
 func (c *linuxContainer) Start(process *Process) error {
 	c.m.Lock()
 	defer c.m.Unlock()
+	// configure linux personality before starting the process
+	if c.config.Personality != nil {
+		err := system.SetLinuxPersonality(c.config.Personality.Domain)
```

Comment:
I'm still unsure this is the best place. @cyphar can you please take a look?

Reply:
Since we want to set personality as soon as process starts so this seems the best place to me.

Reply:
Well, here you set it _before_ the process starts, not _as soon as_.

Reply:
sorry for confusion i meant this should happen just before the process start or as soon as the process start so execution happens in configured execution domain. But i am cool if you think there is any better place for this call.

Reply:
This should probably happen as close to execve as possible -- that would mean in `standard_init_linux.go` and `setns_init_linux.go` -- right before we apply seccomp filters probably.

Reply:
I'm not super familiar with personalities to be honest, my guess is that we would want to do this as close to execve as possible since I don't know what effect setting the personality will have to runc's internal code. As such, I would suggest this should probably happen as close to execve as possible -- that would mean in `standard_init_linux.go` and `setns_init_linux.go` -- right before we apply seccomp filters probably.

@flouthoc When talking about the "process start" -- this code is being run very early in container setup. Do you mean that we need to set this way before the users' code has even started executing (I would've thought we only need to set it right before we execve the users' binary)?

Reply:
@cyphar I actually tried it in `standard_init_linux.go` a while ago for some reasons I don't exactly recall i was having some issues, something something on the lines of goroutines and because of it personality tests were failing. I'll try it again and post insights thanks.